src/test/java/org/junitpioneer/internal/PioneerAnnotationUtilsTests.java

```diff
@@ -553,10 +553,23 @@ void onClassRepeated() {
 			assertThat(result)
 					.hasSize(1)
 					.allSatisfy(annotation -> assertThat(annotation).isInstanceOf(MetaAnnotatedTestAnnotation.class))
-					.extractingResultOf("value")
+					.extracting(this::extractingResultOfValue)
 					.containsExactlyInAnyOrder("Annotated with repeatable 1");
 		}
 
+		// see https://github.com/assertj/assertj/issues/2760
+		String extractingResultOfValue(Annotation annotation) {
+			if (annotation instanceof MetaAnnotatedTestAnnotation) {
+				return ((MetaAnnotatedTestAnnotation) annotation).value();
+			} else if (annotation instanceof NonRepeatableTestAnnotation) {
+				return ((NonRepeatableTestAnnotation) annotation).value();
+			} else if (annotation instanceof RepeatableTestAnnotation) {
+				return ((RepeatableTestAnnotation) annotation).value();
+			} else {
+				return null;
+			}
```

Comment:
```suggestion
		String extractingResultOfValue(Annotation annotation) {
			// can't wait for type patterns!
			if (annotation instanceof MetaAnnotatedTestAnnotation)
				return ((MetaAnnotatedTestAnnotation) annotation).value();
			if (annotation instanceof NonRepeatableTestAnnotation)
				return ((NonRepeatableTestAnnotation) annotation).value();
			if (annotation instanceof RepeatableTestAnnotation)
				return ((RepeatableTestAnnotation) annotation).value();
			return null;
```

Didn't run the tests - is `return null` expected? If not, should we throw an exception?

Reply:
Quickly had a look the AssertJ code. Looks like it is OK for the extractor function to return `null`. We would then get something like "Expected 'Annotated with repeatable 1' but was 'null'".

Nonetheless, it makes the test results harder to debug, so I will throw an exception.